<launch>

  <node pkg="tf2_ros" type="buffer_server" name="tf2_buffer_server" />

  <node pkg="giskard_examples" type="pr2_controller" name="pr2_controller" output="screen">
    <rosparam param="joint_names"
      file="$(find giskard_examples)/config/pr2_left_upper_body.yaml" />
    <param name="controller_description" 
      textfile="$(find giskard_examples)/controller_specs/pr2_left_upper_body_qp_position_control.yaml" />
    <param name="frame_id" value="base_link" />
    <remap from="~joint_states" to="/joint_states" />
  </node>

  <node pkg="giskard_examples" type="interactive_marker" name="interactive_marker" output="screen">
    <param name="frame_id" value="l_gripper_tool_frame" />
    <remap from="~goal_out" to="/pose_stamped_transformer/in" />
  </node>

  <node pkg="giskard_examples" type="pose_stamped_transformer" name="pose_stamped_transformer" output="screen">
    <param name="frame_id" value="base_link" />
    <remap from="~out" to="/pr2_controller/goal" />
  </node>
 
</launch>
